Assignment2/Q1/src/BinaryTree.java

```diff
@@ -0,0 +1,93 @@
+/**
+ * Created by Alice Colt.
+ *
+ * Implementaiton of a Binary Tree with functions for insert, serach and takeAncestors
+ *
+ */
+public class BinaryTree {
+    private NodeBinaryTree root;
+
+    //Constructor
+    public BinaryTree(NodeBinaryTree new_root){
+        root = new_root;
+    }
+
+
+    public void insert(NodeBinaryTree new_node) {
+
+        NodeBinaryTree actualNode = root;
+
+        if (new_node != null) {
+
+            while (actualNode != null) {
+                if (new_node.compareTo(actualNode) == 1 && actualNode.getLeft() == null) {
+                    actualNode.setLeft(new_node);
+                    new_node.setParent(actualNode);
+                    break;
+                }//if
+                else if (new_node.compareTo(actualNode) == -1 && actualNode.getRight() == null) {
+                    actualNode.setRight(new_node);
+                    new_node.setParent(actualNode);
+                    break;
+                }//else if
+                else if (new_node.compareTo(actualNode) == 0) {
+                    System.out.println(" The node added is a duplicate ");
+                    break;
+                }//else if
+                else {
+                    if (new_node.compareTo(actualNode) == 1) {
+                        actualNode = actualNode.getLeft();
+                    }//if
+                    else if(new_node.compareTo(actualNode) == -1) {
+                        actualNode = actualNode.getRight();
+                    }//else if
```

Comment:
I would pull this block up into the section above - so rather than have the cases for "insert on left, left is null", "insert on right, right is null", "duplicate" and "loop left" and "loop right", I'd put all the code to handle the left case together, and all the code to handle the right together. However, this approach is reasonable.